base/version.jl

```diff
@@ -105,10 +105,18 @@ function split_idents(s::AbstractString)
     return tuple(pidents...)::VerTuple
 end
 
-function VersionNumber(v::AbstractString)
+VersionNumber(v::AbstractString) = parse(VersionNumber, v)
```

Comment:
If you order things in the order `tryparse`, `parse`, `VersionNumber` then you are only using what already is defined which might be slightly easier to read.

Reply:
sounds good!